### PR TITLE
Make number of samples in batch signed

### PR DESCRIPTION
**Pull request by klecki**
## Description:
Rename the curr_tensors_size_ to curr_num_tensors_
for more clarity and make it signed as most of the sample-indexing
is signed in DALI.

Make num_samples() signed in both TensorList and TensorVector.

This makes it consistent with TensorListShape behaviour.

### Affected modules and functionalities:
TV, TL

### Key points relevant for the review:
N/A

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/pipeline/data/tensor_vector.h [hunk 1]
```diff
@@ -367,7 +367,7 @@ class DLL_PUBLIC TensorVector {
 
   std::atomic<int> views_count_;
   std::vector<std::shared_ptr<Tensor<Backend>>> tensors_;
-  size_t curr_tensors_size_;
+  int curr_num_tensors_;
```
Comment (JanuszL) [inline dali/pipeline/data/tensor_vector.h:370]: Maybe ssize_t? Or it is an overkill?

**Review by JanuszL — APPROVED** [review]

File: dali/operators/util/property.cc [hunk 2]
```diff
@@ -42,7 +42,7 @@ template <>
 void SourceInfo<GPUBackend>::FillOutput(workspace_t<GPUBackend>& ws) {
   const auto& input = ws.template Input<GPUBackend>(0);
   auto& output = ws.template Output<GPUBackend>(0);
-  for (size_t sample_id = 0; sample_id < input.num_samples(); sample_id++) {
+  for (int sample_id = 0; sample_id < input.num_samples(); sample_id++) {
     auto si = GetSourceInfo(input, sample_id);
     auto output_ptr = output.raw_mutable_tensor(static_cast<int>(sample_id));
```
Comment (mzient) [inline dali/operators/util/property.cc:47]:
```suggestion
    auto output_ptr = output.raw_mutable_tensor(sample_id);
```

Comment (klecki) [reply dali/operators/util/property.cc:47]: done

**Comment by klecki** [comment]
!build

**Review by mzient — APPROVED** [review]
Approved. I leave it as an open question whether we should also fix use of Index or int64 - this happens at some places in the TensorListView.

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Pull request merged**